### PR TITLE
Bug fix 194

### PR DESCRIPTION
**Pull request by aashish-jain**
Fixes for bug #194 . The port number field now gets disabled if the URL already contains the port number.

### CODE REVIEW

File: src/main/react-front-end/src/views/Transfer/EndpointAuthenticateComponent.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ import IconButton from '@material-ui/core/IconButton';
 import DeleteIcon from '@material-ui/icons/Delete';
 
 import {getType, getName, getDefaultPortFromUri, getTypeFromUri} from '../../constants.js';
+import { bgcolor } from '@material-ui/system';
```
Comment (yinyifu) [inline src/main/react-front-end/src/views/Transfer/EndpointAuthenticateComponent.js]: not used

File: src/main/react-front-end/src/views/Transfer/EndpointAuthenticateComponent.js [hunk 2]
```diff
@@ -109,17 +111,41 @@ export default class EndpointAuthenticateComponent extends Component {
...
+	
+	handleUrlChange = name => event => {
+		let url = event.target.value;
+
+		// Count the number of colons (2nd colon means the URL contains the portnumber)
+		let colonCount = 0;
+		for(let i=0; i < url.length; colonCount+=+(':'===url[i++]));
+
+		this.setState({
+			"portNumFeild": colonCount>=2 ? false : true,
```
Comment (yinyifu) [inline src/main/react-front-end/src/views/Transfer/EndpointAuthenticateComponent.js]: Field Spelled wrong

File: src/main/react-front-end/src/views/Transfer/EndpointAuthenticateComponent.js (hunk 1 above)
Comment (aashish-jain) [reply src/main/react-front-end/src/views/Transfer/EndpointAuthenticateComponent.js]: Fixed in the new commit.

File: src/main/react-front-end/src/views/Transfer/EndpointAuthenticateComponent.js (hunk 2 above)
Comment (aashish-jain) [reply src/main/react-front-end/src/views/Transfer/EndpointAuthenticateComponent.js]: Fixed in the new commit.

**Review by yinyifu — APPROVED** [review]
Approved

**Pull request merged**